sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeFileClient.java

```diff
@@ -1135,8 +1138,22 @@ public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStream
      * @throws DataLakeStorageException If a storage service error occurred.
      */
     public DataLakeFileOpenInputStreamResult openInputStream(DataLakeFileInputStreamOptions options, Context context) {
+        Context newContext;
+        options = options == null ? new DataLakeFileInputStreamOptions() : options;
+        if (options.isUpn() != null) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.set("x-ms-upn", options.isUpn() ? "true" : "false");
+            if (context == null) {
+                newContext = new Context(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
+            } else {
+                newContext = context.addData(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
+            }
+        } else {
+            newContext = null;
+        }
```

Comment:
I'd get rid of `newContext` and just work with `context` directly.

```suggestion
        options = options == null ? new DataLakeFileInputStreamOptions() : options;
        if (options.isUpn() != null) {
            HttpHeaders headers = new HttpHeaders();
            headers.set("x-ms-upn", options.isUpn() ? "true" : "false");
            if (context == null) {
                context = new Context(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
            } else {
                context = context.addData(AddHeadersFromContextPolicy.AZURE_REQUEST_HTTP_HEADERS_KEY, headers);
            }
        }
```